### ISSUE TITLE
refactor: use new node protocol imports

### ISSUE THREAD

**Issue by petermetz** [hyperledger/cactus#1654]
Migrate the entire codebase onto the new syntax as explained here:
https://2ality.com/2021/12/node-protocol-imports.html

Not a particularly important thing, but it definitely has benefits to it and therefore we should at least track this in the backlog even if it's low priority.

Also note: We should do the migration on a package per package basis instead of the entire codebase at once in a huge PR because then it will be very difficult to review properly.

**Important**: It must be verified that this does not cause any backward compatibility issues with older versions of NodeJS (if it does, then this needs to be pushed back until the active LTS version supports it by default without any breakage)

---

From the article:

Previously:

    import * as fs from 'fs/promises';

Now:

    import * as fs from 'node:fs/promises';

**Comment by Leeyoungone** [comment]
Hello! May I take this issue up please?

**Comment by petermetz** [comment]
@Leeyoungone Yes, sure thing! First please verify that it is workable on Node 14 as well (because if not then we need to put this on ice until either v14 goes end-of-life or until the community backports this feature to v14 with a feature release.

**Comment by Leeyoungone** [comment]
@petermetz Based on the article that's [linked](https://2ality.com/2021/12/node-protocol-imports.html), I believe that this is workable for `Node 14` and even `Node 12`


<img width="605" alt="Screen Shot 2021-12-20 at 3 06 39 PM" src="https://user-images.githubusercontent.com/85588157/146832880-e544ac79-a630-4069-a7b4-31471b5b360e.png">

**Comment by petermetz** [comment]
@Leeyoungone I think it does answer the question, but the answer it gives is that you'll have to send a separate PR first upgrading the versions in the CI which will be a prerequisite for all other changes ;-) (let's call it PR zero)

Here[1] you see that we are currently using below Node `v14.18.0` so the CommonJS requires can work (that's what Typescript compiles the `import x from y` statements down to in the end)
[1] https://github.com/hyperledger/cactus/blob/main/.github/workflows/ci.yml#L56

> When working per package, were we thinking about one commit per this PR be for one package or opening up 16 different PRs one for each package and linking them all to this issue?

Let's start small and do it on a per package basis first, starting with the smallest/simplest package, e.g. PR one (after PR zero was merged) could be about maybe the common package or the core-api package (whichever is easier, I'm not sure).

---

And here I'd like to doubly call out the most important acceptance criteria again:

> Important: It must be verified that this does not cause any backward compatibility issues with older versions of NodeJS (if it does, then this needs to be pushed back until the active LTS version supports it by default without any breakage)

What this means is that once you are at the point where PR one is ready, we'll need to do a test with a canary release of the package to npm and then try and install that package and use it with something older than NodeJS v14.18.x to have sufficient proof that we are not locking out everyone who uses v14, but an older version of it.

**Comment by Leeyoungone** [comment]
@petermetz Thank you for the reply! It makes things a lot more easier!! 

Just a follow up question: 
> upgrading the versions in the CI which will be a prerequisite for all other changes

I was wondering if we were upgrading the CI to `16.0.0` or `17.3.0` (which is the newest version) or another version. I was going to assume that we wanted it to be `v16.0.0` since that's what mentioned on the linked website but I just wanted to confirm. Thanks again!

**Comment by petermetz** [comment]
@Leeyoungone No worries, thank you as well ;-)
It has to be 16.x.y (so whatever is latest as long as it starts with 16) and the reason for this is because the v16 version is the one that has LTS (long term support) from the community (v17 does not, only the even numbered major versions do in general)

**Comment by Leeyoungone** [comment]
@petermetz Sounds good, thank you so much for the further clarification!

**Comment by outSH** [comment]
@Leeyoungone Any update on this?
